### ISSUE TITLE
Add bash / zsh / fish shell completion for subcommands and flags

### ISSUE THREAD

**Issue by j-lakeman** [federico-terzi/espanso#2115]
### Description

Would be cool if subcommands like `status` and flags like `--help` could be autocompleted upon pressing TAB. This would increase the ease of use of this already great application lots IMHO.

### Motivation

productivity / ease of use

### Alternatives

keep using `--help` and manually typing subcommands and flags

### Have you tried building it? Or would you like it?

- [ ] I'd be willing to contribute this feature to Espanso myself.

**Comment by smeech** [comment]
I agree, the Espanso commands could be much easier to use, which is why I wrote [this](https://pastebin.com/ETdFixy1), producing this [output](https://pastebin.com/iKgxWLgf) and eventually documented them [here](https://espanso.org/docs/command%20lIne/cli_list/), to index them and make them easier to find.

Shell autocompletion is, I suspect, outside the scope of Espanso development, I fear. However, following your suggestion, a little research suggests adding it would be a fairly straightforward, if time-consuming, task, especially as each shell needs different settings.

Whether Espanso commands are used enough to be worth the effort is debatable. :neutral_face:

**Comment by werdahias** [comment]
the autocompletions can be generated automatically by [clap-complete](https://crates.io/crates/clap_complete)

**Comment by AucaCoyan** [comment]
Yes! this is a missing feature I want to deliver anytime. Sadly I couldn't find the time, and only did the completions for my shell (nushell).
But you are right! completions can be automated with `clap`.
